### DISCUSSION TITLE
contributing/development/configuring_an_ide/visual_studio_code

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#46] [User-contributed notes]
# contributing/development/configuring_an_ide/visual_studio_code

Visual Studio Code is a free cross-platform code editor by Microsoft(not to be confused with Visual Studio). Importing the project: Make sure the C/C++ extension is installed. You can find instruct...

https://docs.godotengine.org/en/latest/contributing/development/configuring_an_ide/visual_studio_code.html

**Comment by martinkunc** [comment]
Hi everyone,
it looks like in example of launch.json in the path to godot_gdb_pretty_print instead of `scripts` should be `utility`, like this:
```
source ${workspaceRoot}/misc/utility/godot_gdb_pretty_print.py
```

**Reply by Calinou** [reply]
I've opened a PR to fix this: https://github.com/godotengine/godot-docs/pull/9774

Thanks for reporting it :slightly_smiling_face:

**Comment by edward93** [comment]
Those steps didn't work for me `Godot 4.4-beta.1 C#`. 
This [article](https://gamefromscratch.com/c-development-with-godot-and-visual-studio-code-tutorial/) helped, however.

Leaving this here in case someone else is struggling to debug their project with VS code.

`tasks.json`
```json
  "tasks": [
    {
      "type": "dotnet",
      "task": "build",
      "problemMatcher": ["$msCompile"],
      "group": {
        "kind": "build",
        "isDefault": true
      },
      "label": "build"
    }
  ]
```

`launch.json`
```json
  "configurations": [
    {
      "name": "Run Godot Run",
      "type": "coreclr",
      "request": "launch",
      "program": "C:/Program Files/Godot/Godot_v4.4-beta1_mono_win64/Godot_v4.4-beta1_mono_win64.exe",
      "args": ["--path", "${workspaceRoot}"],
      "stopAtEntry": false,
      "cwd": "${workspaceFolder}",
      "environment": [],
      "console": "internalConsole",
      "visualizerFile": "${workspaceFolder}/platform/windows/godot.natvis",
      "preLaunchTask": "build"
    }
  ]
```

**Reply by LicataT** [reply]
Thank you very much, this helped me alot

**Reply by Calinou** [reply]
This page is about setting up Visual Studio Code for C++ development, not C# development. It's expected that the steps described here will not be optimal for working on a Godot C# project.

C# configuration steps are described here instead: https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_basics.html#configuring-an-external-editor

I've opened a pull request for this:

- https://github.com/godotengine/godot-docs/pull/10761

**Comment by ChennySH** [comment]
I use c# and VS code. Can I upload my Godot project to GitHub using VSCode? And if not how to upload my Godot project?

**Reply by Prospy006** [reply]
Yeah just push the git, and make sure to create a git before you push it.

**Comment by Linshiqi** [comment]
When using vscode on macos with M chip, the launch.json should be like this:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Launch Project",
            "type": "lldb-dap",
            "request": "launch",
            "program": "${workspaceFolder}/bin/godot.macos.editor.dev.arm64",
            "args": [
                "--editor",
                "--path",
                "/path/to/godot/game"
            ],
            "cwd": "${workspaceFolder}",
            "preLaunchTask": "build"
        }
    ]
}
```

**Comment by HeeiYa** [comment]
If SCons is installed inside a Python virtual environment, VS Code tasks do not inherit the activated venv.
Instead of calling scons directly, invoke it via the selected interpreter:
```
"command": "${command:python.interpreterPath}",
"args": ["-m", "SCons"]
```
